lib/CodeGen/CGBuiltin.cpp

```diff
@@ -2426,6 +2426,9 @@ RValue CodeGenFunction::EmitBuiltinExpr(const FunctionDecl *FD,
     break;
   }
   case Builtin::BI_Dynamic_check: {
+    if (!getLangOpts().CheckedC)
+      break;
```

Comment:
This can happen, if we're not in Checked C, and we come across a call to `_Dynamic_check`. However, this `break` falls through to some code that will error for "unrecognized builtin", which is our intended behavior I believe. 

I can add a test that compiling something that includes `_Dynamic_check` but without checked c enabled works as the programmer would expect (calling their self-defined `_Dynamic_check` function)